src/workerd/api/node/i18n.c++

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+
+#include "i18n.h"
+
+#include <workerd/jsg/exception.h>
+
+#include <unicode/putil.h>
+#include <unicode/timezone.h>
+#include <unicode/uchar.h>
+#include <unicode/uclean.h>
+#include <unicode/ucnv.h>
+#include <unicode/udata.h>
+#include <unicode/uidna.h>
+#include <unicode/ulocdata.h>
+#include <unicode/urename.h>
+#include <unicode/ustring.h>
+#include <unicode/utf16.h>
+#include <unicode/utf8.h>
+#include <unicode/utypes.h>
+#include <unicode/uvernum.h>
+#include <unicode/uversion.h>
+
+namespace workerd::api::node {
+
+namespace i18n {
+
+namespace {
+
+struct ConverterDisposer : public kj::Disposer {
+  static const ConverterDisposer INSTANCE;
+  void disposeImpl(void* pointer) const override {
+    ucnv_close(reinterpret_cast<UConverter*>(pointer));
+  }
+};
+
+const ConverterDisposer ConverterDisposer::INSTANCE;
+
+const char* getEncodingName(Encoding input) {
+  switch (input) {
+  case Encoding::ASCII:
+    return "us-ascii";
+  case Encoding::LATIN1:
+    return "iso8859-1";
+  case Encoding::UCS2:
+    return "utf16le";
+  case Encoding::UTF8:
+    return "utf-8";
+  default:
+    KJ_UNREACHABLE;
+  }
+}
+
+typedef kj::Maybe<kj::Array<kj::byte>> (*TranscodeImpl)(kj::ArrayPtr<kj::byte> source,
+                                                        Encoding fromEncoding, Encoding toEncoding);
+
+kj::Maybe<kj::Array<kj::byte>> TranscodeDefault(kj::ArrayPtr<kj::byte> source,
+                                                Encoding fromEncoding, Encoding toEncoding) {
+  Converter to(toEncoding);
+  std::string substitude(to.minSize(), '?');
+  to.setSubstitudeChars(substitude);
+  Converter from(fromEncoding);
+
+  auto limit = source.size() + to.maxSize();
+  auto out = kj::heapArray<kj::byte>(limit);
```

Comment:
I wonder if we shouldn't cap this at a maximum size just to be safe. I know that Node.js does not but....